app/components/channel_drawer/channels_list/channel_item/channel_item.js

```diff
@@ -22,19 +27,48 @@ export default class ChannelItem extends PureComponent {
         fake: PropTypes.bool,
         isUnread: PropTypes.bool,
         mentions: PropTypes.number.isRequired,
+        navigator: PropTypes.object,
         onSelectChannel: PropTypes.func.isRequired,
         status: PropTypes.string,
         type: PropTypes.string.isRequired,
         theme: PropTypes.object.isRequired
     };
 
+    static contextTypes = {
+        intl: intlShape
+    };
+
     onPress = wrapWithPreventDoubleTap(() => {
         const {channelId, currentChannelId, displayName, fake, onSelectChannel, type} = this.props;
         requestAnimationFrame(() => {
             onSelectChannel({id: channelId, display_name: displayName, fake, type}, currentChannelId);
         });
     });
 
+    onPreview = () => {
+        const {channelId, navigator} = this.props;
+        if (Platform.OS === 'ios' && navigator && this.previewRef) {
+            const {intl} = this.context;
+
+            navigator.push({
+                screen: 'ChannelPeek',
+                previewCommit: false,
+                previewView: this.previewRef,
+                previewActions: [{
+                    id: 'action-mark-as-read',
+                    title: intl.formatMessage({id: 'mobile.channel.markAsRead', defaultMessage: 'Mark As Read'})
+                }],
+                passProps: {
+                    channelId
+                }
+            });
+        }
+    };
+
+    setPreviewRef = (ref) => {
+        this.previewRef = ref;
+    };
```

Comment:
You might be able to do:

```javascript
set previewRef(ref) {
    this.previewRef = ref;
};
```

Reply:
I think the names clash with that, but I don't think that's needed any way